sonar-project.properties

```diff
@@ -0,0 +1,62 @@
+sonar.projectKey=unreal-plugin
+sonar.projectName=UnrealPlugin
+sonar.sources=.
+sonar.cplusplus.file.suffixes=.cpp,.h,.hpp
+sonar.cpp.file.suffixes=.cpp,.h,.hpp
+
+# Exclude build directories and third-party code
+sonar.exclusions=**/Build/**,**/Intermediate/**,**/ThirdParty/**,**/Content/**
+
+# Configure C++ analysis
+sonar.cpp.errorRecoveryEnabled=true
+sonar.cpp.microsoft=true
+sonar.cpp.standardLibrary=microsoft
+
+# Ignore Unreal-specific naming conventions
+sonar.issue.ignore.multicriteria=e1,e2,e3,e4,e5,e6
+
+# Custom parameters for Unreal-specific patterns
+sonar.cplusplus.complexity.threshold=15
+sonar.cplusplus.lines.of.code.per.file.threshold=1000
+sonar.cplusplus.maximum.function.parameters.threshold=8
+
+# Allow UPROPERTY macro usage
+sonar.issue.ignore.multicriteria.e1.ruleKey=cpp:S1186
+sonar.issue.ignore.multicriteria.e1.resourceKey=**/*.h
+
+# Allow UFUNCTION macro usage
+sonar.issue.ignore.multicriteria.e2.ruleKey=cpp:S5421
+sonar.issue.ignore.multicriteria.e2.resourceKey=**/*.h
+
+# Allow GENERATED_BODY macro
+sonar.issue.ignore.multicriteria.e6.ruleKey=cpp:S995
+sonar.issue.ignore.multicriteria.e6.resourceKey=**/*.h
+
+# Allow Unreal naming conventions (PascalCase for classes, camelCase for methods)
+sonar.issue.ignore.multicriteria.e3.ruleKey=cpp:S100
+sonar.issue.ignore.multicriteria.e3.resourceKey=**/*.h
+
+# Allow pointer member variables
+sonar.issue.ignore.multicriteria.e4.ruleKey=cpp:S5008
+sonar.issue.ignore.multicriteria.e4.resourceKey=**/*
+
+# Allow Unreal's memory management patterns
+sonar.issue.ignore.multicriteria.e5.ruleKey=cpp:S5025
+sonar.issue.ignore.multicriteria.e5.resourceKey=**/*
+
+# Allow keyword virtual on overrides
+sonar.issue.ignore.multicriteria.e2.ruleKey=cpp:S3471
+sonar.issue.ignore.multicriteria.e2.resourceKey=**/*.h
+
+# Allow keyword virtual on overrides
+sonar.issue.ignore.multicriteria.e2.ruleKey=cpp:S3471
+sonar.issue.ignore.multicriteria.e2.resourceKey=**/*.h
+
+# Allow enum "Collisions" for enum BlueprintType declarations
+sonar.issue.ignore.multicriteria.e2.ruleKey=cpp:S841
+sonar.issue.ignore.multicriteria.e2.resourceKey=**/*.h
+
+# Enable specific rules useful for Unreal development
+sonar.cpp.suffixes.headers=.h,.hpp
+sonar.cpp.suffixes.sources=.cpp
+sonar.c.file.suffixes=
```

Comment:
Could these go in the top block along with `sonar.cpp.file.suffixes=.cpp,.h,.hpp`? They look thematically similar to those.

Reply:
Actually I'm not sure we need both - Trimmed out a bunch, but going to do some experimenting locally, so not looking for another review quite yet